lib/connection.js

```diff
@@ -225,11 +249,15 @@ Connection.prototype.createSocket = function () {
 /**
  * @private
  */
+Connection.prototype.createNewConnection = function() {
+    if (this.deferredConnection === null && this.sockets.length < this.options.maxConnections) {
```

Comment:
I think this method adds unnecessary complexity. The wrapper is not needed. The only other place `createConnection` is called directly (bypassing this check) is on 124 were you first check whether the max number of connections has been reached before then checking whether the number of sockets is less than the minimum. By changing `maintainMinConnection` to

```
Connection.prototype.maintainMinConnection = function() {
    if (typeof (this.options.minConnections - this.sockets.length) === 'number') {
        if (this.options.minConnections > this.sockets.length) {
            return;
        }
        this.createNewConnection();
    }
}
```